spoc/fragments.py

```diff
@@ -0,0 +1,120 @@
+"""This part of spoc is responsible for dealing aligned fragments that
+have not yet been converted to contacts. It deals with label information
+as well as expanding fragments to contacts."""
+
+from typing import Dict, Union
+import pandas as pd
+import dask.dataframe as dd
+import numpy as np
+from itertools import combinations
+from .dataframe_models import FragmentSchema, ContactSchema
+from .contacts import Contacts
+
+
+class Fragments:
+    """Genomic fragments that can be labelled or not"""
+
+    def __init__(self, fragment_frame: Union[pd.DataFrame, dd.DataFrame]) -> None:
+        self._data = FragmentSchema.validate(fragment_frame)
+        self._contains_meta_data = True if "meta_data" in fragment_frame.columns else False
```

Comment:
Tbh, `meta_data` looks strange to me, but I can go with this naming instead of `metadata`.

Reply:
Agreed, google says 400M hits for `metadata` against 100k for `meta_data` ;) I renamed it in 14456454f0ee321606ff4f5677be09235cccb11c